library/src/test/java/com/karumi/headerrecyclerview/HeaderRecyclerViewAdapterBaseTest.java

```diff
@@ -18,22 +18,27 @@
 
 import android.support.v7.widget.RecyclerView;
 import android.view.ViewGroup;
+import com.karumi.headerrecyclerview.testBaseClass.PowermockTest;
 import java.util.List;
 import org.junit.Test;
+import org.powermock.core.classloader.annotations.PrepareForTest;
 
 import static junit.framework.Assert.assertEquals;
 import static org.junit.Assert.assertFalse;
 import static org.junit.Assert.assertTrue;
-import static org.mockito.Mockito.mock;
-import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.never;
 import static org.mockito.Mockito.verify;
+import static org.powermock.api.mockito.PowerMockito.mock;
+import static org.powermock.api.mockito.PowerMockito.spy;
+import static org.powermock.api.mockito.PowerMockito.when;
 
 /**
  * Base test class created to be extended by every HeaderRecyclerViewAdapter test case. Any
  * HeaderRecyclerViewAdapter extension have to match the contract described in this test.
  */
+@PrepareForTest(RecyclerView.ViewHolder.class)
 public abstract class HeaderRecyclerViewAdapterBaseTest<VH extends RecyclerView.ViewHolder, H, T, F>
-    extends RobolectricTest {
+    extends PowermockTest {
```

Comment:
I tried use `android.support.test.runner.AndroidJUnit4` test runner and mockito, run in "Android Instrumentation Test" to mock `android.support.v7.widget.RecyclerView.ViewHolder.getAdapterPosition()`, but not work. It's a final method, and it seems that google do not provide a mockable version of support-v7 library. I look at  [Robolectric Add-On Modules page](http://robolectric.org/using-add-on-modules/), but they do not provide support-v7 mockable version, too. 
I'm not able to refactor the system to introduce a wrapper class that I control, because holder parameter of `onViewRecycled(VH holder)` is passed in by RecyclerView api, and it must be a   `RecyclerView.ViewHolder`. 

So I only have two choices:
1. don't unit test this method.
2. use powermock to mock final method.

And I failed to integrate Powermock and Robolectric together, as I mentioned before." Both Powermock and Robolectric use a custom classloader, and they will have conflicts."
Another error is Powermock will never be able to mock final method when use with Robolectric, it just delegate Powermock.mock() call to Mockito.mock(), and mockito gives me an Exception when I attempt mocking final methods.